File: SoloLearn Codes/Mathematics/Mathematics.java
import java.util.List;
import java.util.Scanner;
import java.util.regex.MatchResult;
import java.util.regex.Pattern;

public class Mathematics {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int num = sc.nextInt();
        sc.nextLine();
        List<String> expressions = Pattern.compile("\\(([^()]+)\\)")
                .matcher(sc.nextLine()).results()
                .map(MatchResult::group).toList();
        List<List<String>> tokens = expressions.stream()
                .map(e -> Pattern.compile("([0-9]+)|([+\\-*/])")
                        .matcher(e).results().map(MatchResult::group)
                        .toList())
                .toList();
        for(int x = 0; x < tokens.size(); x++) {
            int count = Integer.parseInt(tokens.get(x).get(0));
            for(int i = 1; i < tokens.get(x).size(); i += 2) {
                int n = Integer.parseInt(tokens.get(x).get(i + 1));
                String op = tokens.get(x).get(i);
                if(op.equals("+")) count += n;
                if(op.equals("-")) count -= n;
                if(op.equals("*")) count *= n;
                if(op.equals("/")) count /= n;
            }
            if(count == num) {
                System.out.println("index " + x);
                return;
            }
        }
        System.out.println("none");
    }
}

File: SoloLearn Codes/Mathematics/mathematics.swift
import Foundation

let target = Int(readLine()!)!
let input = readLine()!
var idx = "none"
let expressions = try NSRegularExpression(pattern: "\\(([^()]+)\\)")
	.matches(in: input, range: NSRange(input.startIndex..., in: input))
	.map{ String(input[Range($0.range, in: input)!]) }
let tokens = try expressions.map{ x in
	return try NSRegularExpression(pattern: "([0-9]+)|([+\\-*/])")
		.matches(in: x, range: NSRange(x.startIndex..., in: x))
		.map{ String(x[Range($0.range, in: x)!]) }
}
for i in 0..<tokens.count {
	var count = Int(tokens[i][0])!
	for j in stride(from: 1, through: tokens[i].count - 1, by: 2) {
		let num = Int(tokens[i][j + 1])!
		let op = tokens[i][j]
		switch op {
			case "+":
				count += num
				break
			case "-":
				count -= num
				break
			case "*":
				count *= num
				break
			case "/":
				count /= num
				break
			default:
				count += 0
		}
	}
	if count == target {
		idx = "index \(i)"
		break
	}
}
print(idx)